src/util/Xhr.js

```diff
@@ -56,21 +57,38 @@ class Xhr {
         this.version = version;
         this.sharedLink = sharedLink;
         this.sharedLinkPassword = sharedLinkPassword;
+        this.responseInterceptor = responseInterceptor;
         this.axios = axios.create();
 
         if (typeof responseInterceptor === 'function') {
-            this.axios.interceptors.response.use(responseInterceptor);
+            // Called on any non 2xx response
+            this.axios.interceptors.response.use(responseInterceptor, this.errorInterceptor.bind(this));
         }
 
         if (typeof requestInterceptor === 'function') {
             this.axios.interceptors.request.use(requestInterceptor);
         }
     }
 
+    /**
+     * Error interceptor that wraps the passed in responseInterceptor
+     *
+     * @param {Object} error - Error object from axios
+     * @return {Promise} rejected promise with error info
+     */
+    errorInterceptor(error: Object): Promise<any> {
```

Comment:
=> fat arrow so you dont have to bind.
Also, didnt you want to return `Promise.reject(error)` per your prior comment so that you dont want to send our code a potentially modified error object